### PR TITLE
Dice: Backport 0.2.0 face_values format (kinda)

### PR DESCRIPTION
**Pull request by 20kdc**
**Changes**

* face_values can now have the keys and values swapped, as in 0.2.0. Internal representation is a dictionary of values (as strings) to array of Vector3 normals. Array mustn't be empty. The first vector will be used as the rotation when setting the value.
* Removed restriction on value count matching dice sides. (While the total amount of normals could be checked, IMO this 
restriction is arbitrary.)
* Face values are arbitrary; they are now stringified during load. Numbers are parsed when totalling requires it.
* When only a single dice is selected, the Total indicator instead now says Value, and the value given is "raw" (i.e. for named faces, this shows the face name).

**Fixes/Solves**
*What does this PR fix or solve? If applicable, put the issue number here.*

image
This screenshot is an example use-case: Fudge/FATE dice. These have two blanks (value 0), two + (value 1) symbols and two - (value -1) symbols.
The way you calculate rolls with these is to roll four at once, then total the values. Tabletop Club almost supports this perfectly, but without this PR, their values cannot be properly represented (at least without perhaps tricks like int/float differentiation) and thus it's not possible to get a total.

Numbers (as used)
```
face_values = {
    Vector2(0.0, 0.0): 1,
    Vector2(180.0, 0.0): 1,
    Vector2(0.0, 90.0): 0,
    Vector2(0.0, -90.0): 0,
    Vector2(-90.0, 0.0): -1,
    Vector2(90.0, 0.0): -1
}
```

Names (as an example)
```
face_values = {
    Vector2(0.0, 0.0): "+",
    Vector2(180.0, 0.0): "+",
    Vector2(0.0, 90.0): " ",
    Vector2(0.0, -90.0): " ",
    Vector2(-90.0, 0.0): "-",
    Vector2(90.0, 0.0): "-"
}
```

This also mostly helps #209 (though it's not possible to specify a numeric value for a named value).

_**NOTE:** Make sure this PR goes to the `master` branch! If this fix applies to
previously released versions, then the commits will be cherry-picked afterwards._
Done.

### CODE REVIEW

**Comment by drwhut** [comment]
Thank you for the PR! 🥳 

My only concern with it's current implementation is that it would not be compatible with the new face-value system in v0.2.0 (which has already been implemented and tested in the `0.2-rewrite` branch, see [here](https://github.com/drwhut/tabletop-club/blob/664781ab62d4a6d882e3eda67e70c9c51c02a633/game/assets/asset_pack_type_catalog.gd#L570C3-L570C3) for how the game reads the `config.cfg` file on that branch).

For reference, the format for the `face_values` property in v0.2.0 is essentially reversed from what it is in v0.1.x, in that the keys are the rotation required to make the given face point upwards, and the value is the, well, value:
```ini
; An example 'face_values' from v0.2.0:
face_values = {
    Vector2(0.0, 0.0): null,
    Vector2(90.0, 0.0): 2,
    Vector2(0.0, 90.0): 4.5,
    Vector2(180.0, 0.0): "wat",
}
```

I did it this way for two reasons: 1. it solved the duplicate face-value problem, as only the keys are required to be unique, and 2. I figured it is incredibly unlikely that anyone would want two separate values for one face.

I've also made it so that in v0.2.0, the game will still accept the current format for backwards-compatibility, but it will show a warning saying the new format should be used instead.

So yeah to prevent incompatibilites between versions, either the v0.2.0 system will need to be backported, or the v0.2.0 system will need to be replaced with this PR's system.

**Comment by 20kdc** [comment]
I'll go with backporting the v0.2.0 system, now I know what that looks like

**Comment by 20kdc** [comment]
Hopefully this *should* be compatible?

File: docs/custom_assets/asset_packs/asset_types.rst [hunk 1]
```diff
@@ -136,6 +136,7 @@ dice. The total will be shown at the top of the context menu.
 If ``face_values`` is not configured, the dice will always report ``0`` as its
 value.
 
+If the same value needs to be present on multiple faces, a list may be specified rather than a single Vector2, as so:  ``-1: [Vector2(0.0, 0.0), Vector2(180.0, 0.0)]``
```
Comment (drwhut) [inline docs/custom_assets/asset_packs/asset_types.rst:139]: The docs will need to be updated with the new changes.

File: game/Scripts/AssetDB.gd [hunk 2]
```diff
@@ -1234,44 +1234,36 @@ func _import_asset(from: String, pack: String, type: String, config: ConfigFile,
...
-			elif type.ends_with("d6"):
-				num_faces = 6
-			elif type.ends_with("d8"):
-				num_faces = 8
-			elif type.ends_with("d10"):
-				num_faces = 10
-			elif type.ends_with("d12"):
-				num_faces = 12
-			elif type.ends_with("d20"):
-				num_faces = 20
```
Comment (drwhut) [inline game/Scripts/AssetDB.gd:1250]: I'm not so sure if the game should just ignore the number of faces - at the very least, I think the game could show a warning if the number of assigned values does not match, that way the creator can spot the "error" and correct it.

File: game/Scripts/AssetDB.gd [hunk 3]
```diff
@@ -1234,44 +1234,36 @@ func _import_asset(from: String, pack: String, type: String, config: ConfigFile,
...
-					for key in face_values:
+				for key in face_values:
+					var value = face_values[key]
+					# Figure out 0.2.0 A:V versus 0.1.x V:A
+					var fv_angle: Vector2
+					var fv_value: String
+					if key is Vector2:
+						# key is known so don't check it
+						fv_angle = key
+						fv_value = str(value)
```
Comment (drwhut) [inline game/Scripts/AssetDB.gd:1252]: Think it's worth having some type-checking for the value, just to close up any potential exploits. For reference, in v0.2.0 the valid data types are Null, Integers, Floats, and Strings.

**Review by drwhut — CHANGES_REQUESTED** [review]
Looks good! I've just got a few points that need addressing, but once they're resolved I'll test the changes on my end and merge if I can't find anything else :+1:

File: game/Scripts/AssetDB.gd (hunk 2 above)
Comment (20kdc) [reply game/Scripts/AssetDB.gd:1250]: The number of assigned values doesn't really make sense due to faces with the same value. Total normal count would probably work.

Comment (20kdc) [reply game/Scripts/AssetDB.gd]: Implemented as a warning based on total normal count.

File: game/Scripts/AssetDB.gd (hunk 3 above)
Comment (20kdc) [reply game/Scripts/AssetDB.gd:1275]: Added, see f74d4d8

File: docs/custom_assets/asset_packs/asset_types.rst (hunk 1 above)
Comment (20kdc) [reply docs/custom_assets/asset_packs/asset_types.rst]: Oops, probably lost it during the staging process... added in f74d4d8

**Review by drwhut — APPROVED** [review]
Looks good, thank you! Will test as soon as I can before merging :+1:

**Comment by 20kdc** [comment]
Ah, thank you

**Comment by drwhut** [comment]
No, thank YOU! :grin:

**Pull request merged**